src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/TeddyHelper.cs

```diff
@@ -306,7 +306,7 @@ private static (Vector512<byte> Low, Vector512<byte> High) GetNibbles(Vector512<
         [CompExactlyDependsOn(typeof(AdvSimd.Arm64))]
         private static Vector128<byte> Shuffle(Vector128<byte> maskLow, Vector128<byte> maskHigh, Vector128<byte> low, Vector128<byte> high)
         {
-            return Vector128.ShuffleUnsafe(maskLow, low) & Vector128.ShuffleUnsafe(maskHigh, high);
+            return SearchValues.ShuffleNativeModified(maskLow, low) & Vector128.ShuffleNative(maskHigh, high);
```

Comment:
Should the second `Vector128.ShuffleNative(maskHigh,`... be a `SearchValues.ShuffleNativeModified(maskHigh,`...?

This seems to be the pattern everywhere else...

Reply:
I think it's correct as-is, though it's not obvious from this method alone.
It's relying on the fact that `GetNibbles` was used to produce the `low` and `high` vectors here, and only the `low` one isn't guaranteed to be in `[0, 15]` range.

Reply:
Yes, if you look in [this function](https://github.com/hamarb123/runtime/blob/7137ac3010fe03441e8cc97ab16ed44dc6a8b64b/src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/TeddyHelper.cs#L267) (which is called before all the uses), you'll see that the high bits are already normalised, whereas the low bits are not when Ssse3 is available, because it utilises Ssse3's masking behaviour. The behaviour of the treatment of the low bits matches before by explicitly ensuring we get Ssse3 (as we are explicitly depending on its behaviour), but the high bits are guaranteed to be within the required range, so we just call `ShuffleNative`